src/core/ledger/grainIntegration.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import {Ledger, type PayoutAddress, type CurrencyId} from "./ledger";
+import {type Distribution} from "./distribution";
+import {getDistributionBalances} from "./distributionSummary/distributionSummary.js";
+import {type Currency} from "./currency.js";
+import {type IdentityId} from "../identity";
+import * as G from "./grain";
+
+type Transfer = {|
+  amount: G.Grain,
+  memo: string,
+|};
+
+export type AccountDistributions = Map<PayoutAddress, G.Grain>;
+export type PayoutAddressToId = Map<PayoutAddress, IdentityId>;
+export type TransferredGrain = Map<PayoutAddress, Transfer>;
+
+export type IntegrationResult = {|
+  // Amounts that are actually distributed by the integration.
+  // If Grain is still not tracked offchain, these can be recorded as transfers
```

Comment:
"still not tracked offchain" I feel like this is the opposite of what is meant here. -> "still tracked offchain" or "still not tracked onchain"